src/ngraph/op/fused/clamp.cpp

```diff
@@ -18,12 +18,105 @@
 #include "ngraph/builder/make_constant.hpp"
 #include "ngraph/op/maximum.hpp"
 #include "ngraph/op/minimum.hpp"
+#include "ngraph/runtime/reference/clamp.hpp"
+#include "ngraph/util.hpp"
 
 using namespace std;
 using namespace ngraph;
 
 constexpr NodeTypeInfo op::Clamp::type_info;
 
+namespace
+{
+    template <element::Type_t ET, typename T>
+    bool evaluate(const HostTensorPtr& arg, const HostTensorPtr& out, T min, T max, size_t count)
+    {
+        runtime::reference::clamp<T>(
+            arg->get_data_ptr<ET>(), out->get_data_ptr<ET>(), min, max, count);
+        return true;
+    }
+
+    bool evaluate_clamp(
+        const HostTensorPtr& arg, const HostTensorPtr& out, double min, double max, size_t count)
+    {
+        auto ceil_func = [](double x) { return std::ceil(x); };
+        auto floor_func = [](double x) { return std::floor(x); };
```

Comment:
You could just do a simple cast instead of this. For integer types the value stored in the double will be an integer.

Reply:
There is nothing preventing the user from providing non-integral values for min/max when the input is an integer type.  The fact that the op is specified using doubles (attributes) for min/max leaves us with a decision to make in terms of the policy we will use when converting min/max to type T.  Here, by using ceil for min and floor for max, we select the nearest integer values "between" min and max for both min and max.  This may not be the best policy, but it's the one that @diyessi and I discussed and agreed on.

Reply:
I added comments to clamp.cpp documenting the policy we take toward converting min/max to integer types.